hikari/users.py

```diff
@@ -64,52 +64,216 @@ class UserFlag(enums.Flag):
+    """
+
+    DISABLE_PREMIUM = 1 << 21
+    """User's Nitro features have been disabled.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
     ACTIVE_DEVELOPER = 1 << 22
-    """User is an active bot developer."""
+    """User is a developer or team member that owns an app which has had an application command
+    executed in the last 30 days.
+    """
+
+    PROVISIONAL_ACCOUNT = 1 << 23
+    """User is a provisional account used with the social layer integration.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    HIGH_GLOBAL_RATE_LIMIT = 1 << 33
+    """User has their global ratelimit raised to 1,200 requests per second.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    DELETED = 1 << 34
+    """User's account is deleted.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    DISABLED_SUSPICIOUS_ACTIVITY = 1 << 35
+    """User's account is disabled due to suspicious activity and must reset their password.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    SELF_DELETED = 1 << 36
+    """User's account is deleted by the user.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    PREMIUM_DISCRIMINATOR = 1 << 37
+    """User's account has a custom Nitro discriminator.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    USED_DESKTOP_CLIENT = 1 << 38
+    """User has used the desktop client.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    USED_WEB_CLIENT = 1 << 39
+    """User has used the web client.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    USED_MOBILE_CLIENT = 1 << 40
+    """User has used the mobile client.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    DISABLED = 1 << 41
+    """User's account is disabled.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    VERIFIED_EMAIL = 1 << 43
+    """User has verified their email address.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    QUARANTINED = 1 << 44
+    """User's account is quarantined and can't create DMs or join servers.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
+
+    ELIGIBLE_FOR_POMELO_USERNAME_MIGRATION = 1 << 47
+    """User is eligible to migrate their account from using discriminators to using the unique username system.
+
+    !!! note
+        This flag is not documented, but appears stable and consistent.
+    """
```

Comment:
This is not available publically in the API and only available on data harvest dumps (aka, when you request Discord for your information)